config/jobs-chromeos.yaml

```
@@ -379,6 +379,9 @@ _anchors:
         - mainline
         - next
         - kernelci
+        - collabora-chromeos-kernel_for-kernelci
+        - media_master
+        - media_fixes
```

```suggestion
        - collabora-chromeos-kernel
        - media
```

Only trees (and/or branches) can be added there, build configs cannot (yet)

---

I see. I'll update you. Is the master branch being selected by default right now?

---

No, there is no currently no way to add rules for specific combinations of trees/branches. The rules mean basically `run job if $tree in $allowed_trees and $branch in $allowed_branches`.

You can just list the trees, it will only use the branches configured in build_configs:
* `mainline` will use `master`
* `next` will use `master`
* `kernelci` will use `staging-mainline`, `staging-next` and `staging-stable`; 
* `collabora-chromeos-kernel` will use `for-kernelci`
* `media` will use `master` and `fixes`

One thing you could do is the following, if you really want/need to add branch-based rules:

```yaml
    tree:
      - mainline
      - next
      - collabora-chromeos-kernel
      - media
    branch:
      - master
      - fixes
      - for-kernelci
```

This won't prevent from running on e.g. `media/for-kernelci` if it were to exist some day, but is a bit more restrictive than using tree-based rules only. Please note this isn't needed at all for now as we don't have any tree/branch combination that wouldn't match.

Note: wasn't `kernelci` supposed to be removed from the trees list?

---

Created https://github.com/kernelci/kernelci-core/issues/2548 for the feature request